pkg/gds/client/profiles_test.go

```diff
@@ -0,0 +1,196 @@
+	wrongVersion := &client.Profiles{
+		Version: "v2",
+	}
+	data, err := yaml.Marshal(wrongVersion)
+	require.NoError(t, err)
+	_, err = f.Write(data)
+	require.NoError(t, err)
+	require.NoError(t, f.Close())
+	_, err = client.Load()
+	require.Error(t, err)
+	require.NoError(t, os.RemoveAll(profileYAML))
+
+	// Config is valid
+	f, err = os.Create(profileYAML)
+	require.NoError(t, err)
+	expected := client.DefaultProfiles()
+	data, err = yaml.Marshal(expected)
+	require.NoError(t, err)
+	_, err = f.Write(data)
+	require.NoError(t, err)
+	require.NoError(t, f.Close())
+
+	// Load should return the default profiles
+	profiles, err := client.Load()
+	require.NoError(t, err)
+	require.Equal(t, expected, profiles)
+}
+
+// Test that GetActive returns the corect profile.
+func TestGetActive(t *testing.T) {
+	profiles := client.DefaultProfiles()
+
+	// Default to active profile
+	actual, err := profiles.GetActive("")
+	require.NoError(t, err)
+	require.Equal(t, profiles.Profiles[profiles.Active], actual)
+
+	// Profile name does not exist
+	_, err = profiles.GetActive("does-not-exist")
+	require.Error(t, err)
+
+	// Profile name specified
+	actual, err = profiles.GetActive(profiles.Active)
+	require.NoError(t, err)
+	require.Equal(t, profiles.Profiles[profiles.Active], actual)
+}
+
+// Test that SetActive marks the specified profile as active and writes to disk.
+func TestSetActive(t *testing.T) {
+	defer os.RemoveAll(profileYAML)
+	profiles := client.DefaultProfiles()
+	f, err := os.Create(profileYAML)
+	require.NoError(t, err)
+	require.NoError(t, f.Close())
+
+	// Profile name does not exist
+	err = profiles.SetActive("does-not-exist")
+	require.Error(t, err)
+
+	// Find a profile that is not active
+	name := ""
+	for n := range profiles.Profiles {
+		if n != profiles.Active {
+			name = n
+			break
+		}
+	}
+	require.NotEmpty(t, name)
+
+	// Active profile should be changed and new config written to disk
+	err = profiles.SetActive(name)
+	require.NoError(t, err)
+	require.Equal(t, name, profiles.Active)
+	data, err := os.ReadFile(profileYAML)
+	require.NoError(t, err)
+	actual := &client.Profiles{}
+	err = yaml.Unmarshal(data, actual)
+	require.NoError(t, err)
+	require.Equal(t, profiles, actual)
+}
+
+// Test that Save writes the profiles to the specified folder.
+func TestSave(t *testing.T) {
+	defer os.RemoveAll(profileYAML)
+	profiles := client.DefaultProfiles()
+	f, err := os.Create(profileYAML)
+	require.NoError(t, err)
+	require.NoError(t, f.Close())
+
+	// Profiles should be written to the local config if not specified.
+	err = profiles.Save(nil)
+	require.NoError(t, err)
+	actual, err := client.Load()
+	require.NoError(t, err)
+	require.Equal(t, profiles, actual)
+
+	// Profiles should be written to the specified config.
+	tmp, err := ioutil.TempDir("", "config-*")
+	defer os.RemoveAll(tmp)
+	require.NoError(t, err)
+	folder := &configdir.Config{
+		Path: tmp,
+		Type: configdir.Local,
+	}
+	err = profiles.Save(folder)
```

Comment:
It was a good call to have the folder passed to save so that this test would only write to the tempdir.